storage/dataflux/integration_test.go

```diff
@@ -115,22 +120,24 @@ func TestIntegration_NextBatch(t *testing.T) {
 	df := NewLister(c, in)
 	defer df.Close()
 	totalObjects := 0
+	counter := 0
 	for {
+		counter++
 		objects, err := df.NextBatch(ctx)
-		if err != nil && err != iterator.Done {
-			t.Errorf("df.NextBatch : %v", err)
-		}
-		totalObjects += len(objects)
 		if err == iterator.Done {
+			totalObjects += len(objects)
 			break
 		}
-		if len(objects) > in.BatchSize {
-			t.Errorf("expected to receive %d objects in each batch, got %d objects in a batch", in.BatchSize, len(objects))
+		if err != nil {
+			t.Errorf("df.NextBatch : %v", err)
 		}
+		totalObjects += len(objects)
 	}
-	if totalObjects != wantObjects {
-		t.Errorf("expected to receive %d objects in results, got %d objects in results", wantObjects, totalObjects)
-
+	if totalObjects != numObjectsPrefix {
+		t.Errorf("expected to receive %d objects in results, got %d objects in results", numObjectsPrefix, totalObjects)
+	}
+	if counter <= 1 {
```

Comment:
I think this condition will never be hit because the counter is updated at the start of the for loop.

Reply:
Sorry, it will be hit because it's greater than or equal to. So intstead I'll change this to a nit: easier to read this code if the counter updates after each loop rather than at the beginning